pkg/component/observability/opentelemetry/operator/opentelemetryoperator.go

```diff
@@ -0,0 +1,408 @@
+			},
+		},
+	}
+}
+
+func (otel *openTelemetryOperator) roleBinding() *rbacv1.RoleBinding {
+	return &rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      roleName,
+			Namespace: otel.namespace,
+		},
+		Subjects: []rbacv1.Subject{
+			{
+				Kind:      rbacv1.ServiceAccountKind,
+				Name:      name,
+				Namespace: otel.namespace,
+			},
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: rbacv1.GroupName,
+			Kind:     "Role",
+			Name:     roleName,
+		},
+	}
+}
+
+func (otel *openTelemetryOperator) deployment() *appsv1.Deployment {
+	return &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      v1beta1constants.DeploymentNameOpenTelemetryOperator,
+			Namespace: otel.namespace,
+			Labels: utils.MergeStringMaps(getLabels(), map[string]string{
+				resourcesv1alpha1.HighAvailabilityConfigType: resourcesv1alpha1.HighAvailabilityConfigTypeController,
+			}),
+		},
+		Spec: appsv1.DeploymentSpec{
+			RevisionHistoryLimit: ptr.To[int32](2),
+			Replicas:             ptr.To[int32](1),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: getLabels(),
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: utils.MergeStringMaps(getLabels(), map[string]string{
+						v1beta1constants.LabelNetworkPolicyToDNS:              v1beta1constants.LabelNetworkPolicyAllowed,
+						v1beta1constants.LabelNetworkPolicyToRuntimeAPIServer: v1beta1constants.LabelNetworkPolicyAllowed,
+					}),
+				},
+				Spec: corev1.PodSpec{
+					ServiceAccountName: name,
+					PriorityClassName:  otel.values.PriorityClassName,
+					SecurityContext: &corev1.PodSecurityContext{
+						RunAsNonRoot: ptr.To(true),
+						RunAsUser:    ptr.To[int64](65532),
+						RunAsGroup:   ptr.To[int64](65532),
+						FSGroup:      ptr.To[int64](65532),
+					},
+					Containers: []corev1.Container{
+						{
+							Name:            name,
+							Image:           otel.values.Image,
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							Args: []string{
+								"--metrics-addr=127.0.0.1:8080",
+								"--enable-leader-election",
+								"--zap-log-level=info",
+								"--zap-time-encoding=rfc3339nano",
+							},
+							Env: []corev1.EnvVar{
+								{
+									Name:  "ENABLE_WEBHOOKS",
+									Value: "false",
+								},
+								{
+									Name: "NAMESPACE",
+									ValueFrom: &corev1.EnvVarSource{
+										FieldRef: &corev1.ObjectFieldSelector{
+											APIVersion: "v1",
+											FieldPath:  "metadata.namespace",
+										},
+									},
+								},
+								{
+									Name: "SERVICE_ACCOUNT_NAME",
+									ValueFrom: &corev1.EnvVarSource{
+										FieldRef: &corev1.ObjectFieldSelector{
+											APIVersion: "v1",
+											FieldPath:  "spec.serviceAccountName",
+										},
+									},
+								},
+							},
+							Resources: corev1.ResourceRequirements{
+								Requests: corev1.ResourceList{
+									corev1.ResourceCPU:    resource.MustParse("100m"),
```

Comment:
100m may be a bit too high and lead to waste of resources. I would start with lower values, e.g. 10m and adapt later based on the needs. Some docs on the subject: https://github.com/gardener/gardener/blob/master/docs/usage/autoscaling/shoot_pod_autoscaling_best_practices.md#summary